shoop/campaigns/models/catalog_effects.py

```diff
@@ -0,0 +1,82 @@
+# This file is part of Shoop.
+#
+# Copyright (c) 2012-2016, Shoop Ltd. All rights reserved.
+#
+# This source code is licensed under the AGPLv3 license found in the
+# LICENSE file in the root directory of this source tree.
+from django.db import models
+from django.utils.translation import ugettext_lazy as _
+
+from shoop.admin.forms.fields import PercentageField
+from shoop.core.fields import MoneyValueField
+from shoop.core.models import PolymorphicShoopModel
+
+
+class CatalogCampaignEffect(PolymorphicShoopModel):
+    identifier = None
+    model = None
+    admin_form_class = None
+    yields = False  # TODO: maybe remove this
+
+    campaign = models.ForeignKey("CatalogCampaign", related_name='effects', verbose_name=_("campaign"))
+
+    def apply_for_product(self, applying_context):
+        """
+        Applies the effect based on given context or yields a new `OrderLine`
+
+        :type context: `shoop.campaigns.modules.CampaignApplyingContext`
+        :return: yields `OrderLine` or returns a `Price` based on context
+        :rtype: `OrderLine`, `Price`
+        """
+        raise NotImplementedError("Not implemented!")
```

Comment:
Now that the applying context has only 4 variables and one of them is redundant (campaign), maybe it would be better to just pass those variables as arguments and ditch the CampaignApplyingContext.

So this would be:
```python
def apply_for_product(self, context, product, price_info):
    """
    Apply this effect for given product in given context.

    :type context: shoop.core.pricing.PricingContext   # ?? check this
    :type product: shoop.core.models.Product
    :type price_info: shoop.core.pricing.PriceInfo
    :rtype: PriceInfo
    """
    pass
```